pkg/router/controller/factory/factory.go

```diff
@@ -124,6 +127,21 @@ func (factory *RouterControllerFactory) Create(plugin router.Plugin, watchNodes,
 		}, &kapi.Secret{}, secretEventQueue, factory.ResyncInterval).Run()
 	}
 
+	// Check ARP Cache and endpoint list
+	go utilwait.Forever(func() {
+		data, err := ioutil.ReadFile("/proc/sys/net/ipv4/neigh/default/gc_thresh3")
+		if err != nil {
+			glog.Warning("Error reading ARP neighbour information")
+		}
+		endpoints, _ := factory.KClient.Endpoints(factory.Namespace).List(metav1.ListOptions{})
+		items := len(endpoints.Items)
```

Comment:
You could do something like:
```
// Check if the router is shared based on namespace or project labels
epCount := 0
if factory.Namespaces != nil {
    allowedNamespaces := factory.Namespaces.NamespaceNames()
    for ep := range endpoints.Items {
        if allowedNamespaces.Has(ep.Namespace) {
                 epCount = epCount + 1
        }
    }
} else {
    epCount = len(endpoints.Items)
}
```
Moving this logic to existing endpoints watcher could be expensive as it needs to fetch allowed namespaces for every event in case of sharded router.